Branch 0.4 release docs

sameerz opened the pull request:
1. Update documents for 0.4 release.  Removed -SNAPSHOT or changed plugin 0.3->0.4, cudf 0.17->0.18
1. Updated image version for GCP Dataproc.  Tested 0.4 works with the latest Dataproc Spark 3.1.1 image
1. One broken link fix in docs/additional-functionality/rapids-udfs.md

revans2 commented on docs/download.md:
```diff
@@ -4,6 +4,52 @@ title: Download
+## Release v0.4.0
+
+New functionality for the release includes
+* Decimal support up to 64 bit, including reading and writing decimal from Parquet
```
Do we want to say anything about it being disabled by default?

sameerz replied:
I added a note on how to enable decimal.  I also updated some text around the decimal config.

revans2 reviewed (approved):
Just one question

jlowe reviewed (commented):
Not under the docs directory but documentation all the same, there are references to 0.4.0-SNAPSHOT in integration_tests/README.md and tests/README.md as well.  It's not a requirement to update those in this PR, but they should be updated by the time the pom files are updated.

jlowe commented on docs/download.md:
```diff
@@ -4,6 +4,52 @@ title: Download
+
+New functionality for the release includes
+* Decimal support up to 64 bit, including reading and writing decimal from Parquet
+* Ability to call native CUDA or cudf functions from Scala, Java or Hive UDFs
```
```suggestion
* Ability for users to provide GPU versions of Scala, Java or Hive UDFs
```

sameerz replied:
Incorporated feedback.

jlowe commented on docs/download.md:
```diff
@@ -4,6 +4,52 @@ title: Download
+* `array_contains` for list operations
+* `collect_list` and `average` for windowing operations
+* Murmur3 `hash` operation 
+* Reading a v2 datasource when the data being read is an `ArrowColumnVector`
```
```suggestion
* Improved performance when reading from DataSource v2 when the source produces data in the Arrow format.
```

sameerz replied:
Incorporated feedback

sameerz commented:
@jlowe @revans2 @tgravescs I updated the docs, and also added that we support Spark 3.0.2.  

@jlowe I will do a follow up to remove `SNAPSHOT` from integration_tests/README.md and tests/README.md

sameerz commented:
build
